lib/ace/transport_app.rb

```diff
@@ -212,59 +219,32 @@ def self.trusted_facts(certname)
       begin
         body = JSON.parse(request.body.read)
         validate_schema(@schemas["run_task"], body)
-        opts = body['target'].merge('protocol' => 'remote')
 
-        # This is a workaround for Bolt due to the way it expects to receive the target info
-        # see: https://github.com/puppetlabs/bolt/pull/915#discussion_r268280535
-        # Systems calling into ACE will need to determine the nodename/certname and pass this as `name`
-        target = [Bolt::Target.new(body['target']['host'] || body['target']['name'], opts)]
-      rescue ACE::Error => e
-        request_error = {
-          "node": target,
-          "target": target,
-          "action": nil,
-          "object": nil,
-          "status": "failure",
-          "result": {
-            "_error": e.to_h
+        inventory = Bolt::Inventory.empty
+        target_data = {
+          'name' => body['target']['host'] || body['target']['name'] || 'remote',
+          'config' => {
+            'transport' => 'remote'
           }
         }
-        return [400, request_error.to_json]
+        target = [Bolt::Target.from_hash(target_data, inventory)]
+      rescue ACE::Error => e
+        return [400, error_result(e).to_json]
       rescue JSON::ParserError => e
-        request_error = {
-          "node": target,
-          "target": target,
-          "action": nil,
-          "object": nil,
-          "status": "failure",
-          "result": {
-            "_error": ACE::Error.to_h(e.message,
-                                      'puppetlabs/ace/request_exception',
-                                      class: e.class, backtrace: e.backtrace).to_h
-          }
-        }
-        return [400, request_error.to_json]
+        request_error = ACE::Error.new(e.message,
+                                       'puppetlabs/ace/request_exception',
+                                       class: e.class, backtrace: e.backtrace).to_h
+
+        return [400, error_result(request_error).to_json]
       rescue StandardError => e
-        request_error = {
-          "node": target,
-          "target": target,
-          "action": nil,
-          "object": nil,
-          "status": "failure",
-          "result": {
-            "_error": ACE::Error.to_h(e.message,
```

Comment:
Lol, not exacly sure what happened here before? But fairly certain that should be `.new`